### DISCUSSION TITLE
What is Stack / Box aquivalent in Fluent UI V2 and best practice on accessing design tokens

### DISCUSSION THREAD

**Discussion by bubuluxWork** [microsoft/fluentui#32836] [Q&A] [answered]
### **Problem**:
From other UI Libraries, I am used to having some sort of `Box` and/or `Stack` component that provides an API to the underlying tokens system (or a dedicated API like a `useTheme`.) 

In Fluents V2 docs under "design language", they specifically state all sorts of token abstractions for stuff like colors, shadows, spacing, materials, motions, and so on. For example [spacing][1]: How do I access the defined spacing ramp? In MUI you would for example be able to do this: 

```javascript
theme.spacing(2); // `${8 * 2}px` = '16px'
```

Or am I supposed to implement this on my own? I would like to understand the best practices here :(

### What I know so far: 
This [post][2] addresses at least the Stack issue, and I understand that [V1][3] had all of those utility abstractions for layout, etc. But I can't find any hints on the official V2 docs and the react storybook.

Ive come across this section of the [V1-docs][4], where they do exactly what Iam looking for:

```javascript
// via react package
import { getTheme } from '@fluentui/react';

const theme = getTheme();

<div style={{ boxShadow: theme.effects.elevation8 }}>

//via theme package
import { DefaultEffects } from '@fluentui/react';

<div style={{ boxShadow: DefaultEffects.elevation8 }} />;
```

Am I supposed to use this method even for V2? 

Any help would be greatly appreciated, we are at the beginning of using FluentUI and I have no prior knowledge of the ecosystem.

  [1]: https://fluent2.microsoft.design/layout
  [2]: https://stackoverflow.com/questions/61659011/is-the-stack-component-for-fluent-ui-the-implementation-of-the-grid-layout
  [3]: https://developer.microsoft.com/en-us/fluentui#/controls/web/stack
  [4]: https://developer.microsoft.com/en-us/fluentui#/styles/web/elevation

**Comment by mainframev** [comment, marked as answer]
Hi @SBArbeit 👋🏻 

Regarding tokens and their usage, styling in general, please take a look at the:

- [storybook docs](https://react.fluentui.dev/?path=/docs/theme-spacing--docs)
- [styles handbook](https://github.com/microsoft/fluentui/blob/master/docs/react-v9/contributing/rfcs/react-components/styles-handbook.md)
- [theming](https://react.fluentui.dev/?path=/docs/concepts-developer-theming--docs)

Tokens are exposed from `@fluentui/react-components`, so you can use them as following:

```jsx
import { tokens } from '@fluentui/react-components';

const useStyles = makeStyles({
  row: {
    display: 'flex',
    gap: tokens.spacingHorizontalL,
  },
});


const Component = () => {
  const styles = useStyles();

  return (
    <div className={styles.row}>
      <div>1</div>
      <div>2</div>
      <div>3</div>
    </div>
  );
};
```

In v9 we do not provide `Stack` and `StackItem` as in Fluent UI Fabric UI. We have a [shim components](https://react.fluentui.dev/?path=/docs/migration-shims-v8-stackshim--docs) for migration purposes. 

> _The Stack and StackItem components are not included in v9. Providing components that expose equivalent layout properties as CSS flexbox required learning another set of props without adding signficant value._

This quote from docs explains why 🙂

**Reply by bubuluxWork** [reply]
Hey @mainframev, thank you so much! 
I don't know how on earth I missed: 

`import { tokens } from '@fluentui/react-components';`

I've tried so many things and probably forgot to check for `tokens` at some point, my bad. 

Also thank you for the clarification about the `Stack` component :)
